internal/gitstream/owners.go

```diff
@@ -13,15 +11,26 @@ type Owners struct {
 	Component string   `yaml:"component"`
 }
 
-func (o *Owners) getAssignee(ctx context.Context, gc *github.Client, userLogin string) (string, error) {
+func (o *Owners) getAssignee(userLogin string) (string, error) {
 
 	for _, approver := range o.Approvers {
 		if approver == userLogin {
 			return userLogin, nil
 		}
 	}
 
-	// User isn't in m/s OWNER file, select a random owner
+	return "", fmt.Errorf("%s is not in the %s file", userLogin, ownersFile)
+}
```

Comment:
I think we could change that function to use the new generic [`slices.Contains`](https://pkg.go.dev/golang.org/x/exp/slices#Contains) function:
```go
func (o *Owners) approverExists(userLogin string) bool {
	return slices.Contains(o.Approvers, userLogin)
}
```
WDYT?

Reply:
It makes sense if we move towards differentiating between API error and user not found in github but I don't think we should.

Reply:
I don't think the implementation has anything to do with the nature of the error while trying to get the username. It's about checking if a username is in a list.

Reply:
Fixed.